SharpPcap/WinDivert/WinDivertDevice.cs

```diff
@@ -156,29 +182,33 @@ public ulong GetParam(WinDivertParam param)
             return value;
         }
 
-        public void SendPacket(RawCapture p)
-        {
-            if (p is WinDivertCapture packet)
-            {
-                WinDivertAddress address = default;
-                address.IfIdx = packet.InterfaceIndex;
-                address.SubIfIdx = packet.SubInterfaceIndex;
-                address.Flags = packet.Flags;
-                SendPacket(new ReadOnlySpan<byte>(p.Data), address);
-                return;
-            }
-            SendPacket(new ReadOnlySpan<byte>(p.Data));
-        }
-
-        public void SendPacket(ReadOnlySpan<byte> p)
+        public void SendPacket(RawCapture p, ICaptureHeader captureHeader)
         {
-            SendPacket(p, GetAddress(p));
+            SendPacket(new ReadOnlySpan<byte>(p.Data), captureHeader);
         }
 
-        private void SendPacket(ReadOnlySpan<byte> p, WinDivertAddress addr)
+        /// <summary>
+        /// </summary>
+        /// <param name="p"></param>
+        /// <param name="header"></param>
+        public void SendPacket(ReadOnlySpan<byte> p, ICaptureHeader captureHeader)
         {
             ThrowIfNotOpen();
             bool res;
+            WinDivertAddress addr = default;
+
+            if (captureHeader == null)
```

Comment:
It could be non null, but still not a `WinDivertHeader`
```suggestion
            if (!(captureHeader is WinDivertHeader))
```